### PR TITLE
Add integration test for procspy edges.

### LINKED ISSUE #779: Add integration test for procspied connections
To prevents the 0.11 bug from appearing again.

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #779

### CODE REVIEW

File: integration/config.sh [hunk 1]
```diff
@@ -44,58 +36,84 @@ scope_end_suite() {
 	done
 }
 
+# this checks we have a named node in the given view
+has() {
+    local view=$1
```
Comment (paulbellamy) [inline integration/config.sh]: indentation

File: integration/330_application_edge_test.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+weave_on $HOST1 run -dti --name client alpine /bin/sh -c "while true; do \
+	nc nginx.weave.local 80 || true; \
+	sleep 1; \
+done"
+
+wait_for application $HOST1 60 nginx client
+
+has applications $HOST1 nginx
+has applications $HOST1 nc
+has_connection applications $HOST1 nc nginx
```
Comment (paulbellamy) [inline integration/330_application_edge_test.sh]: v nice

File: integration/330_application_edge_test.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+start_suite "Test short connections between processes"
+
+WEAVE_DOCKER_ARGS=$ADD_HOST_ARGS weave_on $HOST1 launch
+scope_on $HOST1 launch
+weave_on $HOST1 run -d --name nginx nginx
+weave_on $HOST1 run -dti --name client alpine /bin/sh -c "while true; do \
+	nc nginx.weave.local 80 || true; \
+	sleep 1; \
+done"
```
Comment (paulbellamy) [inline integration/330_application_edge_test.sh]: This isn't actually that short of a connection, as one it is connected the `nc` will hang for a while.

Comment (tomwilkie) [inline integration/330_application_edge_test.sh]: Not supposed to be a short connection (they don't get applied between processes) - I'm testing procspy, which only looks at long connections.  Have updated comment and turned off conntrack for this test to make it more obvious.

**Comment by tomwilkie** [comment]
@paulbellamy PTAL

**Comment by paulbellamy** [comment]
Ok, so the salient point of these tests is that when conntrack is off the probe is forced to use procspy, correct? if so, let's either mention procspy in the test name, or a comment on why conntrack is disabled.

**Comment by tomwilkie** [comment]
@paulbellamy kinda - we're checking process to process edges, which only come from procspy.  Have added this to test name.

**Pull request merged**